src/btree/bt_sync.c

```diff
@@ -238,14 +238,15 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
         __wt_atomic_store_enum(&btree->syncing, WT_BTREE_SYNC_RUNNING);
 
         /*
-         * Reset the number of obsolete time window pages to let the eviction threads continue
-         * marking the clean obsolete time window pages as dirty once the checkpoint is finished.
+         * Reset the variables associated with the clean up of obsolete time window information so
+         * we can resume the work in case the limits have been reached.
          */
         btree->obsolete_tw_pages = 0;
-        is_hs = WT_IS_HS(btree->dhandle);
+        conn->heuristic_controls.obsolete_tw_btree_count = 0;
```

Comment:
Why this is reset for each btree checkpoint? Should this been done when we start the global checkpoint?

Reply:
That's a good point, it is at the wrong location.

Reply:
Moved in a909c3492930c9aee9ee2298da5b2783cd8b5181